js/_actions.js

```diff
@@ -0,0 +1,147 @@
+// polyfills
+require('babel-polyfill');
+var fetch = require('isomorphic-fetch');
+
+export const SET_MESSAGE = "SET_MESSAGE";
+export function setError(errorMessage) {
+	return {
+		type: SET_MESSAGE,
+		message: errorMessage,
+		message_type: "error"
+	};
+}
+
+export const SET_API_ENDPOINT = "SET_API_ENDPOINT";
+export function setAPIEndpoint(url) {
+	return {type: SET_API_ENDPOINT, url: url};
+}
+
+export const SET_ACTIVE_STEP = "SET_ACTIVE_STEP";
+export function setActiveStep(id) {
+	return {type: SET_ACTIVE_STEP, id};
+}
+
+export const GIT_REQUEST_UPDATE = 'GIT_REQUEST_UPDATE';
```

Comment:
Hmm.. yeah, we should define a standard that follows 1) start request, 2) request successful, 3) request failed.. in the name of not having to have commens.

```
UPDATE_GIT_REQUEST // starting request, lets reducers set isFetching = true
UPDATE_GIT_FAILURE // action to signify failed request, let reducers handle errors and set error message for UI
UPDATE_GIT_SUCCESS // action to signify request success, i.e reducer sets isFetching = false and parses the server response into the state 
```

how would that work with approvals?

```
SEND_APPROVAL_REQUEST
SEND_APPROVAL_SUCCESS
SEND_APPROVAL_FAILURE
```

```
SEND_APPROVAL_REJECT_REQUEST
SEND_APPROVAL_REJECT_SUCCESS
SEND_APPROVAL_REJECT_FAILURE
```

halp!